source/common/http/conn_manager_utility.cc

```diff
@@ -38,8 +38,9 @@ absl::string_view getScheme(absl::string_view forwarded_proto, bool is_ssl) {
 } // namespace
 std::string ConnectionManagerUtility::determineNextProtocol(Network::Connection& connection,
                                                             const Buffer::Instance& data) {
-  if (!connection.nextProtocol().empty()) {
-    return connection.nextProtocol();
+  const std::string& next_protocol = connection.nextProtocol();
```

Comment:
`nextProtocol` returns a string value, so this is a reference to a temporary. Can you just use `std::string` for the type?